### PR TITLE
Feature/handle_metadata

### PR DESCRIPTION
**Pull request by worksofliam**
Enables makefile generation to take metadata into consideration:

* Reading each `Rules.mk` in directories and then applying them to the generated makefile
   * If dependencies are coded, then the `Rules.mk` will override Source Orbit.
   * If attributes are coded, then we will apply them to the build command for the object
* Reading each `ibmi.json` and applying those rules to sources/objects in the same directory, taking `tgtCcsid` into account when applying `TGTCCSID` for regular programs, `COMPILEOPT` for SQL programs and `CCSID` on `CRTSRCPF` for items compiled from temporary members.
   * This PR does not add support for `objlib`.
* Test cases for `Rules.mk` and `.ibmi.json`

### CODE REVIEW

**Comment by worksofliam** [comment]
@edmundreinhardt @irfanshar This PR is now ready. I fixed incorrect spellings and made sure the tests are passing. Thanks!

File: cli/src/builders/make/customRules.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+import * as path from "path";
+import { MakeProject } from ".";
+
+/**
+ * Scan for all rules.mk files and read attributes and custom
+ * dependencies into the targets.
+ */
+export function readAllRules(targets: Targets, project: MakeProject) {
+  // getFiles is case insensitive
+  const rulesFiles = getFiles(targets.getCwd(), `**/rules.mk`);
```
Comment (edmundreinhardt) [inline cli/src/builders/make/customRules.ts:14]: FYI, if you are in a case sensitive file system, the case is `Rules.mk`

**Review by edmundreinhardt — APPROVED** [review]

File: cli/src/builders/make/customRules.ts (hunk 1 above)
Comment (worksofliam) [reply cli/src/builders/make/customRules.ts:14]: the `getFiles` API has a property passed into it to ignore case. That applies on every filesystem is runs on.

Comment (edmundreinhardt) [reply cli/src/builders/make/customRules.ts:14]: OK, that explains why the tests work :-)

**Pull request merged**